core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -108,36 +118,93 @@ public InputStream decompress(InputStream is) throws IOException {
 
   /**
    * Registers a decompressor for both decompression and message encoding negotiation.
+   *
+   * @param d The decompressor to register
+   * @param advertised If true, the message encoding will be listed in the Accept-Encoding header.
    * @throws IllegalArgumentException if another compressor by the same name is already registered.
    */
-  public static final void registerDecompressor(Decompressor d) {
-    Decompressor previousDecompressor = decompressors.putIfAbsent(d.getMessageEncoding(), d);
-    checkArgument(previousDecompressor == null,
-        "A decompressor was already registered: %s", previousDecompressor);
+  @ExperimentalApi
```

Comment:
We shouldn't have this annotation on methods of a class that is already marked `ExperimentalApi` ... it might lead to confusion if we ever introduce a method without the annotation.